simplesat/dependency_solver.py

```diff
@@ -142,6 +143,82 @@ def requirements_are_satisfiable(packages, requirements, modifiers=None):
         return False
 
 
+def satisfy_requirements(packages, requirements, modifiers=None):
+    """ Find a collection of packages that satisfy the requirements.
+
+    Parameters
+    ----------
+    packages : iterable of PackageMetadata
+        The packages available to draw from when satisfying requirements.
+    requirements : list of Requirement
+        The requirements used to identify relevent packages.
+    modifiers : ConstraintModifiers, optional
+        If not None, modify requirements before resolving packages.
+
+    Returns
+    -------
+    tuple of PackageMetadata
+        Return a list of packages that together satisfy all of the
+        `requirements`.
+
+    Raises
+    ------
+    SatisfiabilityError
+        If the `requirements` cannot be satisfied using the `packages`.
+
+    MissingInstallRequires
+        If no packages meet a dependency requirement.
+    """
+    request = Request(modifiers=modifiers)
+    for requirement in requirements:
+        request.install(requirement)
+    repositories = (Repository(packages),)
+    pool = Pool(repositories, modifiers=modifiers)
+    transaction = DependencySolver(pool, repositories, []).solve(request)
+    msg = ("""
+        Unexpected operation in the transaction. This should never occur.
+        Something in simplesat is broken.
+        {!r}""")
+    for op in transaction.operations:
+        # Our installed repository was empty so everything should be an install
+        # operation
+        assert isinstance(op, InstallOperation), msg.format(op)
+    packages = tuple(op.package for op in transaction.operations)
+    return packages
+
+
+def simplify_requirements(packages, requirements):
+    """ Return a reduced, but equivalent set of requirements.
+
+    Parameters
+    ----------
+    packages : iterable of PackageMetadata
+        The packages available to draw from when satisfying requirements.
+    requirements : list of Requirement
+        The requirements used to identify relevent packages.
+
+    Returns
+    -------
+    tuple of Requirement
+        The reduced requirements.
+    """
+
+    needed_packages = packages_from_requirements(packages, requirements)
+    pool = Pool([Repository(packages)])
+    R = InstallRequirement.from_constraints
+    dependencies = set(itertools.chain.from_iterable(
+        pool.what_provides(R(con))
+        for package in needed_packages
+        for con in package.install_requires
+    ))
+    simple_requirements = requirements_from_packages(
+        package
```

Comment:
We don't care what the ordering is, but we do need it to be deterministic. In this case, we'd have to sort the output anyway to hide the unspecified iteration order of the set so we wouldn't really gain anything.